tutorials/general/t8_tutorial_build_cmesh.cxx

```diff
@@ -0,0 +1,532 @@
+   * t8_geometry_c      *linear_geom = t8_geometry_linear_new (3);
+   * t8_cmesh_register_geometry (cmesh, linear_geom);    // Use linear geometry 
+   * 
+   * // 4. Definition of the classes of the different trees
+   * t8_cmesh_set_tree_class (cmesh, 0, T8_ECLASS_TET);
+   * t8_cmesh_set_tree_class (cmesh, 1, T8_ECLASS_TET);
+   * t8_cmesh_set_tree_class (cmesh, 2, T8_ECLASS_PRISM);
+   * t8_cmesh_set_tree_class (cmesh, 3, T8_ECLASS_PRISM);
+   * t8_cmesh_set_tree_class (cmesh, 4, T8_ECLASS_PYRAMID);
+   * t8_cmesh_set_tree_class (cmesh, 4, T8_ECLASS_HEX);
+   *
+   * // 5. Classification of the vertices for each tree
+   * t8_cmesh_set_tree_vertices (cmesh, 0, vertices + 12, 4);
+   * t8_cmesh_set_tree_vertices (cmesh, 1, vertices + 24, 4);
+   * t8_cmesh_set_tree_vertices (cmesh, 2, vertices + 42, 6);
+   * t8_cmesh_set_tree_vertices (cmesh, 3, vertices + 60, 6);
+   * t8_cmesh_set_tree_vertices (cmesh, 4, vertices + 78, 5);
+   * t8_cmesh_set_tree_vertices (cmesh, 5, vertices + 93, 8);
+   *
+   * // 6. Definition of the face neighboors between the different trees
+   * t8_cmesh_set_join (cmesh, 0, 2, 0, 4, 0);
+   * t8_cmesh_set_join (cmesh, 1, 3, 0, 4, 0);
+   * t8_cmesh_set_join (cmesh, 2, 5, 0, 0, 0);
+   * t8_cmesh_set_join (cmesh, 3, 5, 1, 1, 0);
+   * t8_cmesh_set_join (cmesh, 4, 5, 4, 2, 0);
+   *
+   * // 7. Commit the mesh
+   * t8_cmesh_commit (cmesh, comm);
+   *
+   * return cmesh;
+   *  */
+
+  /* In many cases the points of the different trees are dependent of these of the previous trees
+   * or of these of the same tree. Then it is reasonable to define the different vertices dependent 
+   * of each other.
+   * Also the order of the sheme is not fixed. The definition of the face neighboors can be 
+   * before the definition of the vertices.
+   * In the following example the tree classes are defined, then the face neighboors are set.
+   * After that the vertices are defined for each tree separate.
+   *  */
+
+  double              vertices[24];
+  int                 i;
+  t8_geometry_c      *linear_geom = t8_geometry_linear_new (3);
+
+  /* Initialization of the mesh */
+  t8_cmesh_t          cmesh;
+  t8_cmesh_init (&cmesh);
+
+  /*  Definition of the geometry */
+  t8_cmesh_register_geometry (cmesh, linear_geom);       /* Use linear geometry */
+
+  /* Defitition of the classes of the different trees */
+  t8_cmesh_set_tree_class (cmesh, 0, T8_ECLASS_TET);
+  t8_cmesh_set_tree_class (cmesh, 1, T8_ECLASS_TET);
+  t8_cmesh_set_tree_class (cmesh, 2, T8_ECLASS_PRISM);
+  t8_cmesh_set_tree_class (cmesh, 3, T8_ECLASS_PRISM);
+  t8_cmesh_set_tree_class (cmesh, 4, T8_ECLASS_PYRAMID);
+  t8_cmesh_set_tree_class (cmesh, 5, T8_ECLASS_HEX);
+
+  /* Classification of the vertices for each tree */
+  t8_cmesh_set_join (cmesh, 0, 2, 0, 4, 0);
+  t8_cmesh_set_join (cmesh, 1, 3, 0, 4, 0);
+  t8_cmesh_set_join (cmesh, 2, 5, 0, 0, 0);
+  t8_cmesh_set_join (cmesh, 3, 5, 1, 1, 0);
+  t8_cmesh_set_join (cmesh, 4, 5, 4, 2, 0);

```

Comment:
```suggestion
  t8_cmesh_set_join (cmesh, 4, 5, 4, 5, 0);

```

Reply:
If the cmesh is ought to look like the cmesh from the attached picture, the pyramid (at face 4) needs to be joined at hex face 5, I think (since the pyramid is located "on top" of the hex; face 2 corresponds to the "bottom face" of the hex)